src/App.js

```diff
@@ -1,31 +1,122 @@
-import {useState} from 'react';
-import {FaPlusCircle} from 'react-icons/fa';
 import './App.css';
-import Title from './Title';
-import ButtonContainer from './ButtonContainer';
-import Button from './Button';
-import List from './List';
+import {useState, useEffect} from 'react';
+import {FaPlus, FaPlusCircle, FaUndo, FaEdit, FaTrashAlt, FaClipboardList} from 'react-icons/fa';
+import {IconContext} from 'react-icons';
+import Dropdown from './Dropdown';
 import ListItem from './ListItem';
-import Icon from './Icon';
 import Modal from './Modal';
+import AddEditItemModal from './AddEditItemModal';
+import AddEditListModal from './AddEditListModal';
+import { PriorityToNumber } from './ListItem';
+import { initializeApp } from "firebase/app";
+import { getFirestore, getDocs, query, collection, doc, setDoc, updateDoc, deleteDoc, orderBy, serverTimestamp, writeBatch } from "firebase/firestore";
+import { useCollectionData, useDocumentData} from "react-firebase-hooks/firestore";
+import { generateUniqueID } from "web-vitals/dist/modules/lib/generateUniqueID";
+
+const firebaseConfig = {
+    apiKey: "AIzaSyDTdxmHJT6utYagkotNRpMLF-EmRhcSYWw",
+    authDomain: "cs124-lab3-d4101.firebaseapp.com",
+    projectId: "cs124-lab3-d4101",
+    storageBucket: "cs124-lab3-d4101.appspot.com",
+    messagingSenderId: "78965801208",
+    appId: "1:78965801208:web:e72bd3c413b6b54b10bb0d"
+  };
+
+const firebaseApp = initializeApp(firebaseConfig);
+const db = getFirestore(firebaseApp);
+const collectionName = "Users";
+// Eventually will change once we add functionality for multiple users
+// For now, have one global user for all list items
+const userDocumentName = "User";
+const collectionPath = `${collectionName}/${userDocumentName}/`;
 
 // Enum for mode of which list items to show
 const ShowState = {
     All: 'All',
     Completed: 'Completed',
     Uncompleted: 'Uncompleted',
 }
+const ShowArr = [
+    'All',
+    'Completed',
+    'Uncompleted',
+]
+
+// Enum for sort condition
+// First element is field, second element is ascending/descending
+const SortBy = {
+    Created_ASC: "Created_ASC",
```

Comment:
Suggest you have as values here:
Created_ASC: {field: 'created', order: 'ASC'}

State information can be objects, not just strings/booleans, etc.

Then, lines 95-98 not needed, just can use sortState.field, sortState.order

Reply:
We tried doing this but there are some funky issues with turning from display string ("Created Ascending") back to its enum ({field: "created", order: "ASC"}) and vice versa so we kept the enum as a string to make the conversion back and forth easier.